components/brave_ads/browser/ads_service_impl.h

```diff
@@ -158,6 +157,7 @@ class AdsServiceImpl : public AdsService,
 
   // TODO(https://github.com/brave/brave-browser/issues/23974) Decouple
   // notification ad business logic.
+  bool CheckIfCanShowNotificationAds();
```

Comment:
Not sure we really need CheckIf

Reply:
CanShowNotificationAds() is already used in mojom interface